### PR TITLE
Added Dot-object typings

### PR DESCRIPTION
**Pull request by nkovacic**
Added Dot-object typings

### CODE REVIEW

File: dot-object/dot-object.d.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
+// Type definitions for Dot-Object v1.4.1
+// Project: https://github.com/rhalff/dot-object
+// Definitions by: Niko Kovačič <https://github.com/nkovacic>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+
+declare namespace DotObject {
+    interface Dot {
+        new(separator: string): Dot;
```
Comment (RyanCavanaugh) [inline dot-object/dot-object.d.ts]: This isn't correct unless you can chain an arbitrary number of `new`s together, which I doubt. The construct signature should be in its own type so this isn't recursive.

File: dot-object/dot-object.d.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+         * If the source path does not exist (undefined)
+         * the target property will not be set.
+         *
+         * @param {String} source
+         * @param {String} target
+         * @param {Object} obj
+         * @param {Function|Array} mods
+         * @param {Boolean} merge
+         */
+        move(source: string, target: string, obj: any, mods?: Function | Array<Function>, merge?: boolean): void;
```
Comment (RyanCavanaugh) [inline dot-object/dot-object.d.ts]: Everywhere in this file should use a more specific function type instead of `Function`, probably `(arg: any) => any`

Comment (nkovacic) [inline dot-object/dot-object.d.ts]: Fixed in the newest commit. Please review it again :).

File: dot-object/dot-object.d.ts (hunk 1 above)
Comment (nkovacic) [inline dot-object/dot-object.d.ts]: Fixed in the newest commit. Please review it again :).

**Pull request merged**